packages/flutter/lib/src/foundation/tree_diagnostics_mixin.dart

```diff
@@ -2,9 +2,1438 @@
+  List<DiagnosticsNode> getProperties();
+
+  /// Children of this DiagnosticsNode.
+  List<DiagnosticsNode> getChildren();
+
+  String get _separator => showSeparator ? ':' : '';
+
+  @override
+  String toString() {
+    if (style == DiagnosticsTreeStyle.singleLine)
+      return toStringDeep();
+
+    if (name == null || name.isEmpty || showName == false)
+      return header;
+
+    return header.contains('\n') ?
+        '$name$_separator\n$header' : '$name$_separator $header';
+  }
+
+  _TextTreeConfiguration get _textTreeConfiguration {
+    switch (style) {
+      case DiagnosticsTreeStyle.dense:
+        return _denseTextConfiguration;
+      case DiagnosticsTreeStyle.normal:
+        return _normalTextConfiguration;
+      case DiagnosticsTreeStyle.offstage:
+        return _dashedTextConfiguration;
+      case DiagnosticsTreeStyle.whitespace:
+        return _whitespaceTextConfiguration;
+      case DiagnosticsTreeStyle.box:
+        return _boxTextConfiguration;
+      case DiagnosticsTreeStyle.singleLine:
+        return _singleLineTextConfiguration;
+    }
+    return _normalTextConfiguration;
+  }
+
+  /// Text configuration to use to connect this node to a `child`.
+  ///
+  /// The singleLine style is special cased because the connection from the
+  /// parent to the child should be consistent with the parent's style as the
+  /// single line style does not provide any meaningful style for how children
+  /// should be connected to their parents.
+  _TextTreeConfiguration _childTextConfiguration(
+    DiagnosticsNode child,
+    _TextTreeConfiguration textStyle,
+  ) {
+    return (child != null && child.style != DiagnosticsTreeStyle.singleLine) ?
+        child._textTreeConfiguration : textStyle;
+  }
+
+  /// Returns a string representation of this node and its descendants.
+  String toStringDeep([String prefixLineOne = '', String prefixOtherLines = '']) {
+    prefixOtherLines ??= prefixLineOne;
+
+    final List<DiagnosticsNode> children = getChildren();
+    final _TextTreeConfiguration config = _textTreeConfiguration;
+    if (prefixOtherLines.isEmpty)
+      prefixOtherLines += config.prefixOtherLinesRootNode;
+
+    final _PrefixedStringBuilder builder =  new _PrefixedStringBuilder(
+        prefixLineOne, prefixOtherLines);
+
+    if (header == null || header.isEmpty) {
+      if (showName && name != null)
+        builder.write(name);
+    } else {
+      if (name != null && name.isNotEmpty && showName) {
+        builder.write(name);
+        if (showSeparator)
+          builder.write(config.afterName);
+
+        builder.write(config.isNameOnOwnLine || header.contains('\n') ?
+            '\n' : ' ');
+      }
+      builder.prefixOtherLines += children.isEmpty ?
+          config.propertyPrefixNoChildren : config.propertyPrefixIfChildren;
+      builder.write(header);
+    }
+
+    final List<DiagnosticsNode> properties = getProperties().where(
+        (DiagnosticsNode n) => !n.hidden).toList();
```

Comment:
nit: if you break after a a `(` you should break before the matching `)` so that it's trivial to tell how the expression is structured. It's hard to tell at a glance where the `where(` call ends here.